lib/WUI/netdev.c

```diff
@@ -79,136 +81,51 @@ static void tcpip_init_done_callback(void *arg) {
         netif_set_link_callback(&eth0, netif_link_callback);
         netif_set_status_callback(&eth0, netif_status_callback);
     }
+    if (netif_add_noaddr(&wlan0, NULL, espif_init, tcpip_input)) {
+        netif_set_link_callback(&wlan0, netif_link_callback);
+        netif_set_status_callback(&wlan0, netif_status_callback);
+    }
     wui_marlin_client_init();
     osMessagePut(networkMbox_id, EVT_TCPIP_INIT_FINISHED, 0);
 }
 
+static struct netif *get_netif_by_id(uint32_t netdev_id) {
+    switch (netdev_id) {
+    case NETDEV_ETH_ID:
+        return &eth0;
+    case NETDEV_ESP_ID:
+        return &wlan0;
+    default:
+        return NULL;
+    }
+}
```

Comment:
I am a bit afraid of using value as index as it can go out of bounds. It already gain value of 2 in some error cases. Instead I am thinking of using netif pointer as interface identifier. We can either us it directly r compare to &eth and &wlan to implement some custom functionality.

Reply:
Well, I think we are already using it to index some other related structures around here. But OK, let's leave it now as it is.